image.go

```diff
@@ -301,23 +324,34 @@ func (i *image) Unpack(ctx context.Context, snapshotterName string, opts ...Unpa
 		a  = i.client.DiffService()
 		cs = i.client.ContentStore()
 
-		chain    []digest.Digest
-		unpacked bool
+		chain   []digest.Digest
+		applyFn func(rootfs.Layer, []digest.Digest) (bool, error)
 	)
-	snapshotterName, err = i.client.resolveSnapshotterName(ctx, snapshotterName)
-	if err != nil {
-		return err
+	if config.SnapshotterName == "" && len(config.Mounts) == 0 {
```

Comment:
The seem to also not be allowed at the same time right? So we need the opposite check too?